### PR TITLE
[Android] Fix ScrollView visibility issue

### LINKED ISSUE #8015: VerticalScrollBarVisibility=“Always” not working in Xamarin Android
### Description
I have a ScrollView in Xamarin Forms , when set `VerticalScrollBarVisibility` be `Always `  it doesn't work .Even though the content size is over the size of scrollview .

### Steps to Reproduce

1. Create a Xamarin Forms Apllication.
2. In xaml add ScrollView with a content view.
3. Run in Android simulator.

```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d"
             x:Class="App15.MainPage">

        <!-- Place new controls here -->
    <ScrollView>
        <AbsoluteLayout HeightRequest="1000"
                        VerticalOptions="FillAndExpand">
            <Image Source="TeslaBackground.jpg"
                   AbsoluteLayout.LayoutBounds="0,0,1,1"
                   AbsoluteLayout.LayoutFlags="All"
                   Aspect="AspectFill" />
            <StackLayout Orientation="Horizontal"
                     AbsoluteLayout.LayoutFlags="All"
                     AbsoluteLayout.LayoutBounds=".5,1,.5,.1">
                <Button x:Name="btnSignIn"
                                        Text="Sign in" />
                <Button x:Name="btnCreateAccount"
                                        Text="Create Account" />
            </StackLayout>
        </AbsoluteLayout>
    </ScrollView>
</ContentPage>
```

### Actual Behavior

untitled

### Basic Information

- Version with issue:
- Last known good version:
- IDE: Visual Studio 16.3.4 2019 Enterprise
- XamarinForms: 4.2.0.848062
 - Android: 9.0

### Reproduction Link

So link : https://stackoverflow.com/questions/58386555/verticalscrollbarvisibility-always-not-working-in-xamarin

Reproduce sample Link：https://www.dropbox.com/s/fp6cdzaqlidea8g/App15.rar?dl=0

### LINKED ISSUE #9103: ScrollView doesnt work with TimePicker inside it Xamarin Forms
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/875358/scrollview-doesnt-work-with-timepicker-inside-it-x.html)._

---
I have found bug When you have timePicker nested inside ScrollView it doesn't work. How can I replace it, or what is your suggestion to solve it. Error
**Android.Views.InflateException: 'Binary XML file line #20: Binary XML file line #20: Error inflating class android.widget.TimePicker'**

---
#### Visual Studio Feedback System on 1/6/2020, 00:18 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Martin Evans on 1/6/2020, 09:32 AM: 

<p>I have just updated to 16.4 and am now also getting the same problem. For me this happens on a Galaxy S5 phone, but not when using an Android Emulator.</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

In this PR https://github.com/xamarin/Xamarin.Forms/pull/8090 was added the support to be able to manage the visibility of the scrollbar in the ScrollView on Android. However, this PR added a repeated issue, the scrollbar visibility the child elements of the ScrollView when initializing (or refreshing). For this reason, this PR https://github.com/xamarin/Xamarin.Forms/pull/10447 revert original one changes.

This PR recovers the initial changes to allow managing the visibility of the scrollbar in ScrollView but also avoiding the issue with the child elements.

fixed-scrollview-visibility-droid

The problem was in the used Context. When creating `ScrollViewRenderer` we were creating a context that used a custom style (using vertical and horizontal scrollbars). This context was used creating `ScrollViewContainer`. So by default `ScrollViewContainer` had `VerticalScrollBarEnabled` set to true (incorrect!).

### Issues Resolved ### 

- fixes #8015
- fixes #9103

### API Changes ###
 
 None

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

None

#### Before
issue-scrollbar-scrollview
#### After
fixed-scroll-droid

### Testing Procedure ###
Launch Core Gallery. Navigate to the "ScrollView Gallery" to verify that can hide/show the ScrollView toolbar. Also, navigate to the "Material Entry Gallery" (for example) and set the focus on different Entries to validate that the "flickering scrollbar" issue in the ScrollView children's is fixed.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by samhouts** [comment]
@jsuarezruiz Can you see if this PR fixes #10882, and if so, retarget to 4.6.0? Thanks!

File: Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs [hunk 1]
```diff
@@ -32,8 +33,9 @@ public class ScrollViewRenderer : NestedScrollView, IVisualElementRenderer, IEff
 		LayoutDirection _prevLayoutDirection = LayoutDirection.Ltr;
 		bool _checkedForRtlScroll = false;
 
-		public ScrollViewRenderer(Context context) : base(context)
+		public ScrollViewRenderer(Context context) : base(new ContextThemeWrapper(context, Resource.Style.NestedScrollBarStyle))
```
Comment (samhouts) [inline Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs:36]: I think #8090 (which first added this wrapper) caused https://github.com/xamarin/Xamarin.Forms/issues/9103. Waiting for confirmation from customers that reverting that PR resolved the issue. If that's the case, then hopefully we're not reintroducing that problem with this PR. It looks like using the right context for the container is the solution. Let's make sure to share artifacts with the customers on #9103 to make sure that it's not going to be a problem.

Comment (samhouts) [reply Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs:36]: Great news! It looks like this PR actually resolves #9103.

**Comment by jsuarezruiz** [comment]
After the E.Z changes int his PR https://github.com/xamarin/Xamarin.Forms/pull/10893, I am going to refactor this one a little bit to apply the same changes.

File: Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs [hunk 2]
```diff
@@ -90,7 +92,7 @@ public void SetElement(VisualElement element)
 				if (_container == null)
 				{
 					Tracker = new VisualElementTracker(this);
-					_container = new ScrollViewContainer(_view, Context);
+					_container = new ScrollViewContainer(_view, _context);
```
Comment (samhouts) [inline Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs:95]: @jsuarezruiz I think this is the change that fixed #9103, so let's try to keep this if we can.

**Comment by hartez** [comment]
@jsuarezruiz It looks like this might also fix #7629?

File: Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs (hunk 2 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.Android/Renderers/ScrollViewRenderer.cs]: I have changed this by applying changes similar to those applied by E.Z in #10893.

**Comment by jsuarezruiz** [comment]
Applied changes based on the E.Z changes in #10893.

**Comment by samhouts** [comment]
The failing tests on Android concern me. :(

**Comment by jsuarezruiz** [comment]
@samhouts Added the "DO-NOT-MERGE-!!!" label until review the failing tests.

**Comment by rmarinho** [comment]
Failing test doesn't seem related

**Pull request merged**